include/sys/aggsum.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * CDDL HEADER START
+ *
+ * This file and its contents are supplied under the terms of the
+ * Common Development and Distribution License ("CDDL"), version 1.0.
+ * You may only use this file in accordance with the terms of version
+ * 1.0 of the CDDL.
+ *
+ * A full copy of the text of the CDDL should have accompanied this
+ * source.  A copy of the CDDL is also available via the Internet at
+ * http://www.illumos.org/license/CDDL.
+ *
+ * CDDL HEADER END
+ */
+/*
+ * Copyright (c) 2017 by Delphix. All rights reserved.
+ */
+
+#ifndef	_SYS_AGGSUM_H
+#define	_SYS_AGGSUM_H
+
+#include <sys/zfs_context.h>
+
+#ifdef	__cplusplus
+extern "C" {
+#endif
+
+#define	CACHE_LINE_SIZE	64
+
+typedef struct aggsum_bucket {
+	kmutex_t asc_lock;
+	int64_t asc_delta;
+	uint64_t asc_borrowed;
+	uint64_t asc_pad[4]; /* pad out to cache line (64 bytes) */
```

Comment:
This padding size is almost certainly wrong on Linux since kmutex_t's are larger.  There's a nice utility called `pahole` you can use to dump the padding and alignment.

Reply:
It looks like the struct is 64 bytes without any extra work; is there a convenient way to add a padding field that will always pad to cacheline size, or should I just leave it at 64 and hope?

Reply:
in that case probably the best thing to do is just drop the field entirely,  `____cacheline_aligned` will force the desired alignment so there's shouldn't be a need to add an explicit pad.